pcdsdevices/sequencer.py

```diff
@@ -0,0 +1,160 @@
+import logging
+
+from ophyd import Device, EpicsSignal, EpicsSignalRO, Component as Cpt
+from ophyd.status import DeviceStatus, SubscriptionStatus
+from ophyd.utils.epics_pvs import raise_if_disconnected
+from ophyd.flyers import FlyerInterface, MonitorFlyerMixin
+
+logger = logging.getLogger(__name__)
+
+
+class EventSequencer(Device, MonitorFlyerMixin, FlyerInterface):
```

Comment:
I'm comparing this with the sequencer we made for sxr.

I notice that there's no way to access the beam owner name or ID. Is this functionality put somewhere else?

There's no :CURSTP pv which tells the current step in the sequencer. Is this intentional?

There's no components for the PVs in the right hand column of the EDM screen for sequencers. These are the synchronization settings.

There's no interface for getting beam rate. Has this been moved to a better place?

Reply:
Stepping in to note that `CURSTP` is included and that beam rate is actually a different IOC, though admittedly we haven't made an object for it.

Reply:
Thanks for comparing! I do have `CURSTP` but I am missing the rest. I'll add all these in.

Reply:
Argh. Ownership PVs have a different prefix. Adding sync settings then calling it good.

Reply:
I'm surprised that the ownership PVs have a different prefix. Are they tied to the beam rate ioc or something completely different?

Reply:
I think there is some base `EventSequencer` that is referenced for these `ECS:SYS:0`. This is where the beam rate and ownership come from. Other EventSequencers have `ECS:SYS:{HUTCH_NO}`. I guess if they are constant for all the sequencers I could just put them in as FormattedComponents and tell it not to add prefixes?

Reply:
I decided to check, because I also have a computer. This is True for the beam owner, but not for the hutch owner. I think we should include `HUTCH_ID` for sure, but I'm 50/50 on doing the `ECS:SYS:0` PVs.
```
$ caget ECS:SYS0:{0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16}:HUTCH_ID
Channel connect timed out: some PV(s) not found.
ECS:SYS0:0:HUTCH_ID            *** not connected
ECS:SYS0:1:HUTCH_ID            1
ECS:SYS0:2:HUTCH_ID            2
ECS:SYS0:3:HUTCH_ID            3
ECS:SYS0:4:HUTCH_ID            4
ECS:SYS0:5:HUTCH_ID            5
ECS:SYS0:6:HUTCH_ID            6
ECS:SYS0:7:HUTCH_ID            7
ECS:SYS0:8:HUTCH_ID            3
ECS:SYS0:9:HUTCH_ID            7
ECS:SYS0:10:HUTCH_ID           2
ECS:SYS0:11:HUTCH_ID           3
ECS:SYS0:12:HUTCH_ID           4
ECS:SYS0:13:HUTCH_ID           0
ECS:SYS0:14:HUTCH_ID           0
ECS:SYS0:15:HUTCH_ID           0
ECS:SYS0:16:HUTCH_ID           5

$ caget ECS:SYS0:{0,1,2,3,4,5,6,7,8,9,10,11,12,13,14,15,16}:BEAM_OWNER_ID
Channel connect timed out: some PV(s) not found.
ECS:SYS0:0:BEAM_OWNER_ID       2
ECS:SYS0:1:BEAM_OWNER_ID       *** not connected
ECS:SYS0:2:BEAM_OWNER_ID       *** not connected
ECS:SYS0:3:BEAM_OWNER_ID       *** not connected
ECS:SYS0:4:BEAM_OWNER_ID       *** not connected
ECS:SYS0:5:BEAM_OWNER_ID       *** not connected
ECS:SYS0:6:BEAM_OWNER_ID       *** not connected
ECS:SYS0:7:BEAM_OWNER_ID       *** not connected
ECS:SYS0:8:BEAM_OWNER_ID       *** not connected
ECS:SYS0:9:BEAM_OWNER_ID       *** not connected
ECS:SYS0:10:BEAM_OWNER_ID      *** not connected
ECS:SYS0:11:BEAM_OWNER_ID      *** not connected
ECS:SYS0:12:BEAM_OWNER_ID      *** not connected
ECS:SYS0:13:BEAM_OWNER_ID      *** not connected
ECS:SYS0:14:BEAM_OWNER_ID      *** not connected
ECS:SYS0:15:BEAM_OWNER_ID      *** not connected
ECS:SYS0:16:BEAM_OWNER_ID      *** not connected
```

Reply:
Maybe it's a good candidate for the beam stats class, beam owner seems closely tied to beam stats. The beam stats class will have tons of ugly PVs from different sources anyway.